googletest/include/gtest/gtest.h

```diff
@@ -2116,9 +2116,14 @@ GTEST_API_ AssertionResult DoubleLE(const char* expr1, const char* expr2,
 // of the dummy variable name, thus allowing multiple SCOPED_TRACE()s
 // to appear in the same block - as long as they are on different
 // lines.
-#define SCOPED_TRACE(message) \
+#define SCOPED_TRACE(...) GTEST_VA_SELECT_(SCOPED_TRACE, __VA_ARGS__) 
+#define SCOPED_TRACE_1(message) \
```

Comment:
in order to avoid a regression in the current behaviour, it would be something like

```c++
#define SCOPED_TRACE(...) \
    ScopedTrace derived_name = ScopedTrace::factory(__FILE__, __LINE__, __VA_ARGS__);
```

```diff
  // The d'tor pops the info pushed by the c'tor.
  //
  // Note that the d'tor is not virtual in order to be efficient.
  // Don't inherit from ScopedTrace!
  ~ScopedTrace();

+ static ScopedTrace factory(const char* caller_file, int caller_line, const Message& message) {
+     return ScopedTrace(caller_file, caller_line, message);
+ }
+
+ static ScopedTrace factory(const char* caller_file, int caller_line, const char* file, int line,
+                                            const Message& message) {
+     return ScopedTrace(file, line, message);
+ }
- private:
-  GTEST_DISALLOW_COPY_AND_ASSIGN_(ScopedTrace);
```

which would probably work, but the is less strict WRT copy/assign. Allowing the move constructor might be a good workaround for that